### ISSUE TITLE
multilevelImpl function fatal error

### ISSUE THREAD

**Issue by mfrisoli126** [ctlab/fgsea#94]
I'm using fgsea through the clusterProfiler package, and I can't figure out why most gene lists work fine while some cause errors that do not produce a result.  It seems to me that the `fgseaMultilevel` function, associated `multilevelImpl` function, and it's use of parallel processing is the source of the error.

Here are downloadable function inputs that were derived from 1 of my real data gene lists: 
```
# Here are 2 function inputs that were derived from 1 of my real data gene lists:
multilevelPathwaysList = https://www.dropbox.com/s/x7q8usq7t1e939s/multilevelPathwaysList.Rdata?dl=0
stats = https://www.dropbox.com/sh/4exhy95mapi4ahq/AADB9ChuMm7AzfxJF-2byP7pa?dl=0

# Here are function arguments that were previously defined:
sampleSize = 101
seed = 50141253
eps = 1e-10
sign = FALSE
BPPARAM =BiocParallel:::SnowParam(detectCores()-1, progressbar = TRUE) # because I'm using windows

cpp.res <- fgsea:::multilevelImpl(multilevelPathwaysList, stats, 
                            sampleSize, seed, eps, sign = sign, BPPARAM = BPPARAM)
```
Error message:
```
  |===========================================                                                                                       |  33%
Error: BiocParallel errors
  element index: 1, 2, 3, 4, 5, 6, ...
  first error: object 'eps' not found
In addition: Warning messages:
1: In serialize(data, node$con) :
  'package:stats' may not be available when loading
2: In serialize(data, node$con) :
  'package:stats' may not be available when loading
3: In serialize(data, node$con) :
  'package:stats' may not be available when loading
```

Furthermore, if I try to run the function without parallel processing, R crashes with a fatal error about 9 times out of 10 times that I try to run the following code:

```
  cpp.res <- fgsea:::multilevelImpl(multilevelPathwaysList, stats, 
                            sampleSize, seed, eps, sign = sign, BPPARAM = BiocParallel::SerialParam())
```

I opened an issue within the clusterProfiler github page as well: https://github.com/YuLab-SMU/clusterProfiler/issues/329

Please help! I have no clue why certain gene lists cause an error, and how parallel processing is somehow linked to this issue?

**Comment by assaron** [comment]
@mfrisoli126 thanks for the report, I can reproduce the error and will look into it. For now, as a workaround, you can switch to `fgseaSimple` algorithm (can be called directly, or by specifying `nperm` option to `fgsea()`)

**Comment by mfrisoli126** [comment]
@assaron Thank you so much!!! I am a PhD student who is new to coding, and this error really stumped me.

I'll try fgseaSimple in the meantime, but I really appreciate you looking into this! 

Best,
Mike

**Comment by assaron** [comment]
This problem should be fixed in version 1.19.2: available on github right now and should be available in Bioconductor-devel branch in a few days.
